samples/hip_sycl_interop_no_buffers/onemkl_gemm_wrapper_no_buffers/onemkl_gemm_wrapper.cpp

```diff
@@ -79,45 +79,63 @@ int onemkl_gemm(sycl::queue& my_queue, float* A, float* B, float* C, int m,
 }
 
 // Run GEMM test via oneMKL
-int oneMKLGemmTest(uintptr_t* nativeHandlers, const char* hip_backend, float* A, float* B, float* C,
-                   int M, int N, int K, int ldA, int ldB, int ldC, float alpha,
-                   float beta) {
+int oneMKLGemmTest(uintptr_t *nativeHandlers, const char *hip_backend, float *A,
+                   float *B, float *C, int M, int N, int K, int ldA, int ldB,
+                   int ldC, float alpha, float beta) {
   std::string hipBackend(hip_backend);
   sycl::queue sycl_queue;
-  if (!hipBackend.compare("opencl")){
+  if (!hipBackend.compare("opencl")) {
     // handle openCl case here
     cl_platform_id hPlatformId = (cl_platform_id)nativeHandlers[0];
     cl_device_id hDeviceId = (cl_device_id)nativeHandlers[1];
     cl_context hContext = (cl_context)nativeHandlers[2];
     cl_command_queue hQueue = (cl_command_queue)nativeHandlers[3];
-    sycl::platform sycl_platform = sycl::opencl::make_platform((pi_native_handle)hPlatformId);
-    sycl::device sycl_device = sycl::opencl::make_device((pi_native_handle)hDeviceId);
-    sycl::context sycl_context = sycl::opencl::make_context((pi_native_handle)hContext);
-    sycl_queue = sycl::opencl::make_queue(sycl_context, (pi_native_handle)hQueue);
+    sycl::platform sycl_platform =
+        sycl::opencl::make_platform((pi_native_handle)hPlatformId);
+    sycl::device sycl_device =
+        sycl::opencl::make_device((pi_native_handle)hDeviceId);
+    sycl::context sycl_context =
+        sycl::opencl::make_context((pi_native_handle)hContext);
+    sycl_queue =
+        sycl::opencl::make_queue(sycl_context, (pi_native_handle)hQueue);
   } else {
```

Comment:
Maybe check that the backend is indeed level zero here, and add an else to crash in case we got neither opencl or level zero.